modules/dynamic/src/smithy4s/dynamic/internals/DynamicModelCompiler.scala

```diff
@@ -210,6 +210,60 @@ private[dynamic] object Compiler {
     override def blobShape(id: ShapeId, shape: BlobShape): Unit =
       update(id, shape.traits, bytes)
 
+    override def enumShape(id: ShapeId, shape: EnumShape): Unit = {
+      val values: List[(Document, EnumValue[Document])] =
```

Comment:
Smithy 1.x-style enums have a runtime representation of Int (`intValue` of the entry). I think we can do the same in 2.x-style enums because the intValue is always unique within an enum.